Source/CombatExtended/Harmony/Harmony_ResearchPrerequisitesUtility.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using HarmonyLib;
+using RimWorld;
+using Verse;
+
+namespace CombatExtended
+{
+    [HarmonyPatch(typeof(ResearchPrerequisitesUtility), nameof(ResearchPrerequisitesUtility.UnlockedDefsGroupedByPrerequisites))]
+    public class Harmony_ResearchPrerequisitesUtility
+    {
+        public static List<Pair<ResearchPrerequisitesUtility.UnlockedHeader, List<Def>>> Postfix(List<Pair<ResearchPrerequisitesUtility.UnlockedHeader, List<Def>>> input)
+        {
+            foreach (var pair in input)
+            {
+                pair.second.RemoveWhere(x => x is AmmoDef ammoDef && ammoDef.menuHidden);
+            }
+            input.RemoveWhere(x => x.second.NullOrEmpty());
```

Comment:
What's exactly the point of removing empty research unlock headers? So that if research ends up unlocking nothing, that it wouldn't have the text "Unlock:" with nothing after?

Reply:
![image](https://github.com/user-attachments/assets/fe4e129f-a579-42a5-a102-91dd5c0d5031)
yeah, pretty much, the heavy weapons research unlocks the 105mm shells, which were disabled, but the header still shows up.